packages/editor/src/components/post-taxonomies/hierarchical-term-selector.js

```diff
@@ -41,333 +32,320 @@ const DEFAULT_QUERY = {
+	 *
+	 * @param {Object} term Term object.
+	 * @return {Promise} A promise that resolves to save term object.
+	 */
+	const addTerm = ( term ) => {
+		return saveEntityRecord( 'taxonomy', slug, term );
+	};
+
+	/**
+	 * Update terms for post.
+	 *
+	 * @param {number[]} termIds Term ids.
+	 */
+	const onUpdateTerms = ( termIds ) => {
+		editPost( { [ taxonomy.rest_base ]: termIds } );
+	};
+
+	/**
+	 * Handler for checking term.
+	 *
+	 * @param {number} termId
+	 */
+	const onChange = ( termId ) => {
 		const hasTerm = terms.indexOf( termId ) !== -1;
 		const newTerms = hasTerm
 			? without( terms, termId )
 			: [ ...terms, termId ];
-		onUpdateTerms( newTerms, taxonomy.rest_base );
-	}
+		onUpdateTerms( newTerms );
+	};
 
-	onChangeFormName( event ) {
+	const onChangeFormName = ( event ) => {
 		const newValue =
 			event.target.value.trim() === '' ? '' : event.target.value;
-		this.setState( { formName: newValue } );
-	}
-
-	onChangeFormParent( newParent ) {
-		this.setState( { formParent: newParent } );
-	}
-
-	onToggleForm() {
-		this.setState( ( state ) => ( {
-			showForm: ! state.showForm,
-		} ) );
-	}
-
-	findTerm( terms, parent, name ) {
-		return find( terms, ( term ) => {
-			return (
-				( ( ! term.parent && ! parent ) ||
-					parseInt( term.parent ) === parseInt( parent ) ) &&
-				term.name.toLowerCase() === name.toLowerCase()
-			);
-		} );
-	}
-
-	onAddTerm( event ) {
+		setFormName( newValue );
+	};
+
+	/**
+	 * Handler for changing form parent.
+	 *
+	 * @param {number|''} parentId Parent post id.
+	 */
+	const onChangeFormParent = ( parentId ) => {
+		setFormParent( parentId );
+	};
+
+	const onToggleForm = () => {
+		setShowForm( ! showForm );
+	};
+
+	const onAddTerm = async ( event ) => {
 		event.preventDefault();
-		const { onUpdateTerms, taxonomy, terms, slug } = this.props;
-		const { formName, formParent, adding, availableTerms } = this.state;
 		if ( formName === '' || adding ) {
 			return;
 		}
 
 		// check if the term we are adding already exists
-		const existingTerm = this.findTerm(
-			availableTerms,
-			formParent,
-			formName
-		);
+		const existingTerm = findTerm( availableTerms, formParent, formName );
 		if ( existingTerm ) {
 			// if the term we are adding exists but is not selected select it
 			if ( ! some( terms, ( term ) => term === existingTerm.id ) ) {
-				onUpdateTerms(
-					[ ...terms, existingTerm.id ],
-					taxonomy.rest_base
-				);
-			}
-			this.setState( {
-				formName: '',
-				formParent: '',
-			} );
-			return;
-		}
-
-		this.setState( {
-			adding: true,
-		} );
-		this.addRequest = apiFetch( {
-			path: `/wp/v2/${ taxonomy.rest_base }`,
-			method: 'POST',
-			data: {
-				name: formName,
-				parent: formParent ? formParent : undefined,
-			},
-		} );
-		// Tries to create a term or fetch it if it already exists
-		const findOrCreatePromise = this.addRequest.catch( ( error ) => {
-			const errorCode = error.code;
-			if ( errorCode === 'term_exists' ) {
```

Comment:
Looks like all this logic is gone, maybe it's fine though I don't see when this could be useful.

Reply:
It was to get the terms again when it failed to add them.
I decided that was unnecessary, what do you think?

Reply:
I think it's unnecessary yes, I think originally the component was loading a trimmed list of terms meaning the potential for conflicts was higher.